app/src/main/java/org/dhis2/usescases/datasets/dataSetTable/dataSetSection/DataValuePresenter.kt

```diff
@@ -0,0 +1,741 @@
+            repository.isCompleted(orgUnitUid, periodId, attributeOptionCombo)
+                .subscribeOn(schedulerProvider.io())
+                .observeOn(schedulerProvider.ui())
+                .subscribe(
+                    view::setCompleteReopenText
+                ) { Timber.e(it) }
+        )
+    }
+
+    private fun initTable() {
+        disposable.add(
+            repository.getCatCombo(sectionName)
+                .flatMapIterable<CategoryCombo> { categoryCombos -> categoryCombos }
+                .map<Sextet<CategoryCombo,
+                        List<DataElement>,
+                        Map<String, List<List<Pair<CategoryOption, Category>>>>,
+                        List<DataSetTableModel>,
+                        List<DataElementOperand>,
+                        List<DataElementOperand>>>
+                { categoryCombo ->
+                    Flowable.zip<CategoryCombo,
+                        List<DataElement>,
+                        Map<String, List<List<Pair<CategoryOption, Category>>>>,
+                        List<DataSetTableModel>,
+                        List<DataElementOperand>,
+                        List<DataElementOperand>,
+                        Sextet<CategoryCombo,
+                            List<DataElement>,
+                            Map<String, List<List<Pair<CategoryOption, Category>>>>,
+                            List<DataSetTableModel>,
+                            List<DataElementOperand>,
+                            List<DataElementOperand>>>(
+                        Flowable.just<CategoryCombo>(categoryCombo),
+                        repository.getDataElements(categoryCombo, sectionName),
+                        repository.getCatOptions(sectionName, categoryCombo.uid()),
+                        repository.getDataValues(
+                            orgUnitUid,
+                            periodTypeName,
+                            periodId,
+                            attributeOptionCombo,
+                            sectionName
+                        ),
+                        repository.getGreyFields(sectionName),
+                        repository.compulsoryDataElements,
+                        Function6<CategoryCombo,
+                            List<DataElement>,
+                            Map<String, List<List<Pair<CategoryOption, Category>>>>,
+                            List<DataSetTableModel>,
+                            List<DataElementOperand>,
+                            List<DataElementOperand>,
+                            Sextet<CategoryCombo,
+                                List<DataElement>,
+                                Map<String, List<List<Pair<CategoryOption, Category>>>>,
+                                List<DataSetTableModel>,
+                                List<DataElementOperand>,
+                                List<DataElementOperand>>>
+                        { val0, val1, val2, val3, val4, val5 ->
```

Comment:
val0, val1...val5 I don't know what are these variables.
Not for now but we will have to check this together :)